hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -194,77 +192,102 @@ case class HoodieFileIndex(spark: SparkSession,
    * @param queryFilters list of original data filters passed down from querying engine
    * @return list of pruned (data-skipped) candidate base-files' names
    */
-  private def lookupCandidateFilesInColStatsIndex(queryFilters: Seq[Expression]): Try[Option[Set[String]]] = Try {
-    val indexPath = metaClient.getColumnStatsIndexPath
+  private def lookupCandidateFilesInMetadataTable(queryFilters: Seq[Expression]): Try[Option[Set[String]]] = Try {
     val fs = metaClient.getFs
+    val metadataTablePath = HoodieTableMetadata.getMetadataTableBasePath(basePath)
```

Comment:
Is the plan to deprecate existing column stats under `.hoodie/.colstatsindex` and remove all usage of it in 0.11?  If not, should we have two modes where metadata col stats index is used when metadata table is enabled, and `.hoodie/.colstatsindex` is used if metadata table is disabled?

Reply:
I also need clarification on how `.hoodie/.colstatsindex` is generated.  Does that come from clustering or it is also updated per write?

Reply:
Yeah, current bespoke implementation will be removed in a follow-up. It's currently updated after clustering completes.

Reply:
When you say "current bespoke implementation will be removed in a follow-up", is this before or after 0.11.0 release?  I think we still need to keep `.hoodie/.colstatsindex` and the data skipping logic based on that, and there should be a flag to choose how data skipping is done between that vs MT col stats.  Because, if user doesn't choose to enable MT col stats in 0.11.0 and there is no data skipping logic based on `.hoodie/.colstatsindex`, data skipping cannot be done unless user goes back to 0.10.x.  The old logic can be removed one release after 0.11.0.

Reply:
Discussed offline: Bespoke implementation of Col Stats Index would be removed in 0.11